HIP/hip-0000-introduction-of-blocks.md

```diff
@@ -0,0 +1,223 @@
+
+| Property | Computed By | Description  |
+| --- | --- | --- |
+| number | Services | Stored in services state (only the current number) and exported in the record stream. It is the consecutive number of the record file that is being incremented by 1 for every new record file. The number will be initially set in services through the bootstrapping process. It is exposed to the 1) EVM during Transaction Execution through the NUMBER opcode; 2) as a new property in the record file and ingested by mirror nodes. |
+| timestamp | Services | Stored in services state (only the last block timestamp) and computed by services. It is the consensusTimestamp of the first transaction in the record file. It is exposed to the 1) EVM during Transaction Execution through the TIMESTAMP opcode; 2) implicitly exported in the record file through the TS of the first transaction in the record file. |
+| hash | Services | Stored in services (last 256 blocks). It is the 32 byte prefix of the runningHash of the previous record file. That is the running hash of the last record stream object from the previous record file. It is exposed to the 1) EVM during Transaction Execution through the BLOCKHASH opcode; 2) In the record file as the End Object Running Hash |
+| baseFeePerGas | Mirror Node | Always zero, since there is no EIP-1559 style floating block capacity fees in Hedera. |
+  | difficulty | Mirror Node | Hardcoded by Mirror Node(s) to hex encoded 0 |
+  | extraData | Mirror Node | Hardcoded by Mirror Nodes(s) to 0x |
+  | gasLimit | Mirror Node |  Computed by Mirror Node(s). The gas throttle limit per second multiplied by the target block time. |
+  | gasUsed | Mirror Node | Computed by Mirror Node(s). The sum of the gasUsed value for all ContractCall and ContractCreate transactions within the block. |
+  | logsBloom | Mirror Node | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array
+(`0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347`) |
+  | miner | Mirror Node | Hardcoded by the Mirror Nodes to the 0.0.98 address in the long zero prefix format.  |
+  | mixHash | Mirror Node | Hardcoded by Mirror Node(s) to 0x |
+  | nonce | Mirror Node | Hardcoded by Mirror Node(s) to hex-encoded 0 |
+  | parentHash | Mirror Node | The hash of the previous block. |
+  | receiptsRoot | N/A | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array 
+(`0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347`). It is expected to be used and populated in the future once traceability work is finalised.  |
+  | sha3Uncles | Mirror Node | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array (`0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347`) |
+  | size | Mirror Node | Computed by Mirror Node(s). The size of the record file. |
+  | stateRoot | Mirror Node | Hardcoded by Mirror Node(s) to the SHA3 computation of empty array (`0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347`) |
```

Comment:
Why SHA3 empty array for this and `receiptsRoot` instead of `0x` like `extraData`? It makes sense for `sha3Uncles` since it's a SHA3, but shouldn't these others just be empty array `0x`?

Reply:
updated